libraries/AP_Scripting/examples/RM3100_self_test.lua

```diff
@@ -0,0 +1,107 @@
+-- Runs the Built-In Self Test on the RM3100 LR circuits
+-- Note COMPASS_DISBLMSK should have the 16th bit set to 1 (RM3100)
+---@diagnostic disable: need-check-nil
```

Comment:
Cheating again ;)

Reply:
Threw in an actual nil check :+1:. But it makes me wonder when `get_device()` actually returns nil. Every return before the `AP_HAL__I2CDevice` is created is associated with an error which would result in the script erroring out rather than really returning nil right?

Reply:
Yeah, you are correct. In which case you can remove the nil check and update the docs here:
https://github.com/ArduPilot/ardupilot/blob/62d9b14a65f2276534ae17306e531f67fcc0cdd8/libraries/AP_Scripting/docs/docs.lua#L138

To just be:

```
---@return AP_HAL__I2CDevice_ud
```